### PR TITLE
Use #cloneNode to clone FixedLayer body

### PR DESCRIPTION
**Pull request by jridgewell**
Re: https://github.com/ampproject/amphtml/pull/6706#discussion_r92881310

Also removes a [temporary fix](https://github.com/ampproject/amphtml/issues/4097) that is [no longer necessary](https://github.com/ampproject/amphtml/pull/6033).

### CODE REVIEW

File: src/service/fixed-layer.js [hunk 1]
```diff
@@ -608,9 +608,9 @@ export class FixedLayer {
       return this.fixedLayer_;
     }
     const doc = this.ampdoc.win.document;
-    const bodyId = doc.body.id;
-    this.fixedLayer_ = doc.createElement('body');
-    this.fixedLayer_.id = bodyId || '-amp-fixedlayer';
+    this.fixedLayer_ = doc.body.cloneNode();
```
Comment (dvoytenko) [inline src/service/fixed-layer.js]: Can you add explicit `/*deep*/ false` here, to make it self documenting?

File: src/service/fixed-layer.js [hunk 2]
```diff
@@ -608,9 +608,9 @@ export class FixedLayer {
...
+    this.fixedLayer_.id += ' -amp-fixedlayer';
```
Comment (dvoytenko) [inline src/service/fixed-layer.js]: This breaks the old fix. We need the actual body.id if one was specified.

File: src/service/fixed-layer.js (hunk 1 above)
Comment (jridgewell) [inline src/service/fixed-layer.js]: Commented.

File: src/service/fixed-layer.js (hunk 2 above)
Comment (dvoytenko) [inline src/service/fixed-layer.js]: Yeah, I can see that. But that breaks the previous fix. All `#bodyid .x` will be broken by this. It should be `fixedLayer_.id = doc.body.id || '-amp-fixedlayer'`. Or, tbh, we can simply kill `-amp-fixedlayer` and always use body.id.

Comment (jridgewell) [inline src/service/fixed-layer.js]: Oh man, I though `id` was like `class`. 😳

Fixed.

Comment (dvoytenko) [inline src/service/fixed-layer.js]: Cool. Do the push and I'll approve.

Comment (jridgewell) [inline src/service/fixed-layer.js]: Done.

File: src/service/fixed-layer.js [hunk 3]
```diff
@@ -608,9 +608,8 @@ export class FixedLayer {
       return this.fixedLayer_;
     }
     const doc = this.ampdoc.win.document;
-    const bodyId = doc.body.id;
-    this.fixedLayer_ = doc.createElement('body');
-    this.fixedLayer_.id = bodyId || '-amp-fixedlayer';
+    this.fixedLayer_ = doc.body.cloneNode(/* deep */ false);
+    this.fixedlayer.removeAttribute('style');
```
Comment (camelburrito) [inline src/service/fixed-layer.js]: we should also be making class = "" ?

**Pull request merged**